frontend/src/components/core/Editor.tsx

```diff
@@ -1,9 +1,61 @@
-import React from 'react';
-import MonacoEditor, { MonacoEditorProps } from 'react-monaco-editor';
+import React, { useState } from 'react';
+import MonacoEditor from 'react-monaco-editor';
+import styled from 'styled-components';
+
+type LanguageType = {
+  [key: string]: {
+    name: string,
+    defaultCode: string,
+  }
+};
+
+export const languages: LanguageType = {
```

Comment:
Could the `defaultCode` be updated whenever the user types new things?

This could just be for later, but I'm thinking that if the user types some things in Java then briefly switches to another language, their old code shouldn't be deleted.

This could also be implemented when other things like saving code even after refresh are built.

Reply:
That's a great idea - just made the change :)